langchain/llms/gpt4all.py

```diff
@@ -0,0 +1,182 @@
+"""Wrapper for the GPT4All model."""
+from typing import Any, Dict, List, Mapping, Optional, Set
+
+from pydantic import BaseModel, Extra, Field, root_validator
+
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+
+
+class GPT4All(LLM, BaseModel):
+    r"""Wrapper around GPT4All language models.
+
+    To use, you should have the ``pyllamacpp`` python package installed, the
+    pre-trained model file, and the model's config information.
+
+    Example:
+        .. code-block:: python
+            from langchain.llms import GPT4All
+            model = GPT4All(model="./models/gpt4all-model.bin", n_ctx=512, n_threads=8)
+
+            # Simplest invocation
+            response = model("Once upon a time, ")
+    """
+
+    model: str
+    """Path to the pre-trained GPT4All model file."""
+
+    n_ctx: int = Field(512, alias="n_ctx")
```

Comment:
Why is alias needed here? I thought it was for reserved fields?

Reply:
@hinthornw ?